src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -218,6 +220,8 @@
                                         String.format("cdap/delta/%s/", context.getApplicationName()),
                                         context, executorService);
     this.baseRetryDelay = baseRetryDelay == null ? 10L : baseRetryDelay;
+    String maxClusteringColumnsStr = context.getRuntimeArguments().get("gcp.bigquery.max.clustering.columns");
+    this.maxClusteringColumns =  maxClusteringColumnsStr == null ? null : Integer.valueOf(maxClusteringColumnsStr);
```

Comment:
nit: remove extra space before maxClusteringColumnsStr

Reply:
This should default to 4 if nothing is set. If the user explicitly sets it to 0 or less, then we can skip the clustering.

Reply:
I have thought about this, initially I was thinking, we will also need to change the default 4 in the feature if bigquery has bumped up the limit. I took another look of this, I think we can use 4 as default like you said, in this case, like you said, we will always had some baseline improvements.  Will update this default logic!